src/System.Device.Gpio.Tests/GpioControllerTestBase.cs

```diff
@@ -150,6 +150,29 @@ void callback(object sender, PinValueChangedEventArgs pinValueChangedEventArgs)
             }
         }
 
+        [Fact]
+        public void AddCallbackFallingEdgeNotDetectedTest()
+        {
+            ManualResetEvent mre = new ManualResetEvent(false);
+            bool wasCalled = false;
+            using (GpioController controller = new GpioController(GetTestNumberingScheme(), GetTestDriver()))
+            {
+                controller.OpenPin(InputPin, PinMode.Input);
+                controller.OpenPin(OutputPin, PinMode.Output);
+                controller.Write(OutputPin, PinValue.Low);
+                controller.RegisterCallbackForPinValueChangedEvent(InputPin, PinEventTypes.Falling, callback);
+                controller.Write(OutputPin, PinValue.High);
+                mre.WaitOne(TimeSpan.FromSeconds(5));
```

Comment:
2 things:
- WaitOne returns bool so you don't need additional bool flag
- Test seem to always last 5 seconds - this seems quite too long IMO - 100ms seems reasonably enough